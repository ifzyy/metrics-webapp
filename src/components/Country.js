import React from 'react';
import PropTypes from 'prop-types';
import { useNavigate } from 'react-router-dom';
import '../styles/country.css';

const Country = ({
  id, country, countryCode,
}) => {
  const navigate = useNavigate();
  const handleClick = () => {
    navigate(`/statistics/${country}`);
  };

  return (

    <li className="country">
      <p className="countryName">{country}</p>
      <p className="countryCode">{countryCode}</p>
      <button type="button" id={id} name={country} onClick={handleClick} className="forward">See details</button>
    </li>

  );
};

Country.propTypes = {
  id: PropTypes.string.isRequired,
  country: PropTypes.string.isRequired,
  countryCode: PropTypes.string.isRequired,
};

export default Country;
