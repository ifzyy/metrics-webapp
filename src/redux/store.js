import { configureStore, combineReducers } from '@reduxjs/toolkit';
import thunk from 'redux-thunk';
import countriesReducer from './fetch/fetch';

const rootReducer = combineReducers({
  countries: countriesReducer,
});

const store = configureStore({
  reducer: rootReducer,
  middleware: [thunk],
});

export default store;
