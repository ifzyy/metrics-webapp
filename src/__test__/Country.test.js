import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../redux/store';
import Country from '../components/Country';

it('Render <Countries /> correctly', () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <MemoryRouter>
          <Country id="80" country="Nigeria" countryCode="NG"/>
        </MemoryRouter>
      </Provider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
