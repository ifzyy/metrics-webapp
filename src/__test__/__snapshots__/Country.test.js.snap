// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render <Countries /> correctly 1`] = `
<li
  className="country"
>
  <p
    className="countryName"
  >
    Nigeria
  </p>
  <p
    className="countryCode"
  >
    NG
  </p>
  <button
    className="forward"
    id="80"
    name="Nigeria"
    onClick={[Function]}
    type="button"
  >
    See details
  </button>
</li>
`;
